editor/src/clj/editor/scene.clj

```diff
@@ -989,21 +989,28 @@
   (run [app-view] (when-let [view (active-scene-view app-view)]
                     (stop-handler view))))
 
-(defn set-camera! [camera-node start-camera end-camera animate?]
-  (if animate?
-    (let [duration 0.5]
-      (ui/anim! duration
-                (fn [^double t]
-                  (let [t (- (* t t 3) (* t t t 2))
-                        cam (c/interpolate start-camera end-camera t)]
-                    (g/transact
-                      (g/set-property camera-node :local-camera cam))))
-                (fn []
-                  (g/transact
-                    (g/set-property camera-node :local-camera end-camera)))))
-    (g/transact
-      (g/set-property camera-node :local-camera end-camera)))
-  nil)
+(defn set-camera!
+  ([camera-node start-camera end-camera animate?]
+   (set-camera! camera-node start-camera end-camera animate? nil))
+  ([camera-node start-camera end-camera animate? on-animation-end]
+   (if animate?
+     (let [duration 0.5]
+       (g/transact (g/set-property camera-node :animating true))
```

Comment:
Maintaining this state was necessary to fix issues related to spamming camera-realign, and also some flickering issues of the toolbar buttons.